Adding copyright to albums #1838

ThanasisMpalatsoukas opened the pull request:
Fixes #1838

Bringing the data of copyright to the frontend and adding a new route on api.php to edit the copyright value.
for issue #1838. After getting this PR accepted i will work on the frontend side and push on frontend the rest

ThanasisMpalatsoukas committed:
feat: added copyright on base_album

Bringing the data of copyright to the frontend and adding a new route
on api.php to edit the copyright value.

ThanasisMpalatsoukas committed:
fix: fixed some spelling mistakes.

ildyria commented:
Thank you for your pull request. :)

Could you please add two tests:
- one where someone adds a copyright
- one where someone removes the copyright.

Also what should be the default value when it is not present?

You may also consider adding some localization strings: Like Copyright, Set copyright etc.. in the `lang/xx/` folders.

ildyria commented on app/Http/Controllers/AlbumController.php:
```diff
@@ -186,6 +187,21 @@ public function setDescription(SetAlbumDescriptionRequest $request): void
+	 */
+	public function setCopyright(SetAlbumCopyrightRequest $request): void
+	{
+		$request->album()->copyright = $request->copyright();
```
You need to add a `@property` annotation to the BaseAlbumImpl and BaseAlbum description.

ThanasisMpalatsoukas replied:
Will look into the above issues. I was thinking the default value would be null or empty string when there is not value present.

ildyria commented on app/Http/Requests/Album/SetAlbumCopyrightRequest.php:
```diff
@@ -0,0 +1,38 @@
+	 */
+	protected function processValidatedValues(array $values, array $files): void
+	{
+		$this->album = $this->albumFactory->findBaseAlbumsOrFail(
```
```suggestion
		$this->album = $this->albumFactory->findBaseAlbumOrFail(
```

ThanasisMpalatsoukas committed:
Update app/Http/Requests/Album/SetAlbumCopyrightRequest.php

Co-authored-by: Benoît Viguier <ildyria@users.noreply.github.com>

ThanasisMpalatsoukas commented:
@ildyria 
Should i use google translate for the translations that i don't know or should i create the keys on all files and leave them empty?

ThanasisMpalatsoukas committed:
fix: added basic localization, properties and tests

tested for editing the copyright value. Creating a different
endpoint for removing copyright might not be necessary as the
copyright value can be edited to be "".

ildyria commented:
> @ildyria Should i use google translate for the translations that i don't know or should i create the keys on all files and leave them empty?

add the keys (the tests will require it anyway) and don't bother with google translate, just put the English. If someone wants to do the translation for their language, we will take the PR :)
I will probably make a change on yours to support French (as it is my native language).

ildyria commented on lang/fr/lychee.php:
```diff
@@ -151,6 +151,8 @@
 	'ALBUM_ABOUT' => 'À propos',
 	'ALBUM_BASICS' => 'Informations de base',
 	'ALBUM_TITLE' => 'Titre',
+	'ALBUM_COPYRIGHT' => '',
```
```suggestion
	'ALBUM_COPYRIGHT' => 'Droits d’auteur',
```

ThanasisMpalatsoukas committed:
Update lang/fr/lychee.php

Co-authored-by: Benoît Viguier <ildyria@users.noreply.github.com>

ildyria commented on lang/fr/lychee.php:
```diff
@@ -151,6 +151,8 @@
 	'ALBUM_BASICS' => 'Informations de base',
 	'ALBUM_TITLE' => 'Titre',
+	'ALBUM_COPYRIGHT' => '',
+	'ALBUM_SET_COPYRIGHT' => '',
```
```suggestion
	'ALBUM_SET_COPYRIGHT' => 'Définir les droits d’auteur',
```

ildyria committed:
Update lang/fr/lychee.php

ThanasisMpalatsoukas committed:
fix: added copyright in langs that didn't have the key

ThanasisMpalatsoukas committed:
Merge branch '1838' of https://github.com/ThanasisMpalatsoukas/Lychee into 1838

ThanasisMpalatsoukas committed:
fix: added default english on all langs

ildyria committed:
Merge branch 'master' into 1838

ildyria commented:
@ThanasisMpalatsoukas  I fixed the conflicts with master.

We will merge this PR when:
- tests are added (see above)
- front-end supports it:
    - You will need to make a PR on Lychee-front for that
    - Once merged, we can rebuild the js files and we will be able to merge this one in master.
    
    
:smiley:

ildyria commented:
@ThanasisMpalatsoukas any updates ? :(

ThanasisMpalatsoukas commented:
@ildyria Hey, was busy with work and stuff. Will probably take a look at it this week :)

ildyria commented:
> was busy with work and stuff.

I know that feeling quite well. :')

ildyria commented:
@ThanasisMpalatsoukas any updates ? :(

ARMeeru commented:
@ildyria Is it still pending?

ildyria commented:
Well yes. Those points have still not been resolved.

> We will merge this PR when:
> 
> * tests are added (see above)
> * front-end supports it:
>   
>   * You will need to make a PR on Lychee-front for that
>   * Once merged, we can rebuild the js files and we will be able to merge this one in master.

Though what I would suggest is to fork this Pull request and start integrating against the Livewire branch of #1303 as it will make it significantly easier for the user to add tests and not have to care for the JS front side.

ARMeeru commented:
I'm interested on this. While finalising that interest, I would like to ping @ThanasisMpalatsoukas in case he's on this but forgot.

ThanasisMpalatsoukas commented:
@ARMeeru thank you for being considerate and pinging me on this, sadly i have too personal work and am unable to continue with this at this point in time. You can go on and finalize this feature :)

ildyria commented:
@ARMeeru If you decide to go on this feature, I strongly advise you to have a look at https://github.com/LycheeOrg/Lychee/pull/2035 before.

You will probably want to check the following files:
https://github.com/LycheeOrg/Lychee/blob/livewire/app/View/Components/Gallery/Album/Details.php
https://github.com/LycheeOrg/Lychee/blob/livewire/resources/views/components/gallery/album/details.blade.php

https://github.com/LycheeOrg/Lychee/blob/livewire/app/Livewire/Components/Forms/Album/Properties.php
https://github.com/LycheeOrg/Lychee/blob/livewire/resources/views/livewire/forms/album/properties.blade.php

And associated tests:
https://github.com/LycheeOrg/Lychee/blob/livewire/tests/Livewire/Forms/Album/PropertiesTest.php

This should be a good read too: https://lycheeorg.github.io/docs/livewire.html

ildyria committed:
Merge branch 'master' into add-copyright-per-album

ildyria committed:
fix migration not properly done.

ildyria committed:
a bit too fast

ildyria committed:
add copyright to details

ildyria committed:
full support of per-album-copyright

ildyria committed:
fix missing langauge

ildyria committed:
fix phpstan

ildyria reviewed (approved):
Tested LGTM.
